### PR TITLE
Bugfix for #10407: Alternative way to access room_id for TikTok lives

### LINKED ISSUE #10407: [TikTok] Live extraction fails sometimes due to new JSON structure
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

From debugging it appears they changed some data in `_real_extract(...)` in `class TikTokLiveIE`, leading to `UserNotLive` exception despite being live. This happens because the room_id can not be extracted properly.

new data dict appears to be structured as follows:
`
{
{'LiveRoom': {...},
'CurrentRoom': {...}
}
`

and the roomId can be found via `(traverse_obj(data, ('LiveRoom', "liveRoomUserInfo", "user", "roomId"), get_all=False)))`

NOTE: Strangely enough my program was working occasionally, meaning there might also be other factors involved.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
Not applicable.
```

### PR DESCRIPTION
**Pull request by mokrueger**
### Fixes https://github.com/yt-dlp/yt-dlp/issues/10407

Adds the described (backwards compatible) way to access the room_id for TikTok lives.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/tiktok.py [hunk 1]
```diff
@@ -457,7 +457,7 @@ def extract_addr(addr, add_meta={}):
             'thumbnails': thumbnails,
             'duration': (traverse_obj(video_info, (
                 (None, 'download_addr'), 'duration', {functools.partial(int_or_none, scale=1000)}, any))
-                or traverse_obj(music_info, ('duration', {int_or_none}))),
+                         or traverse_obj(music_info, ('duration', {int_or_none}))),
```
Comment (bashonly) [inline yt_dlp/extractor/tiktok.py:460]: revert
```suggestion
                or traverse_obj(music_info, ('duration', {int_or_none}))),
```

File: yt_dlp/extractor/tiktok.py [hunk 2]
```diff
@@ -1458,8 +1458,11 @@ def _real_extract(self, url):
 
         if webpage:
             data = self._get_sigi_state(webpage, uploader or room_id)
+
+            # first traverse_obj appears to be deprecated, but remains for now to ensure backwards compatability
             room_id = (traverse_obj(data, ('UserModule', 'users', ..., 'roomId', {str_or_none}), get_all=False)
                        or self._search_regex(r'snssdk\d*://live\?room_id=(\d+)', webpage, 'room ID', default=None)
+                       or (traverse_obj(data, ('LiveRoom', 'liveRoomUserInfo', 'user', 'roomId'), get_all=False))
                        or room_id)
```
Comment (bashonly) [inline yt_dlp/extractor/tiktok.py:1466]:
```suggestion
            room_id = (
                traverse_obj(data, ((
                    ('LiveRoom', 'liveRoomUserInfo', 'user'),
                    ('UserModule', 'users', ...)), 'roomId', {str}, any))
                or self._search_regex(r'snssdk\d*://live\?room_id=(\d+)', webpage, 'room ID', default=room_id))
```

**Review by bashonly — CHANGES_REQUESTED** [review]

**Review by bashonly — APPROVED** [review]

**Comment by mokrueger** [comment]
@bashonly thanks for the quick update. And apologies for the minor formatting issues

**Comment by bashonly** [comment]
@mokrueger all good, thanks for the PR

**Pull request merged**